--- BackgroundWorker.swift.orig	2021-02-10 22:50:58.000000000 +0900
+++ BackgroundWorker.swift	2021-02-10 22:51:44.000000000 +0900
@@ -28,9 +28,15 @@ class BackgroundWorker: NSObject {
                     return
                 }
                 while (!th.isCancelled) {
+                    let t1 = Date()
                     RunLoop.current.run(
                         mode: .default,
                         before: Date.distantFuture)
+                    let t2 = Date()
+                    let delta = t2.timeIntervalSince(t1)
+                    if delta < 0.01 {
+                        Thread.sleep(forTimeInterval: 0.02)
+                    }
                 }
                 Thread.exit()
             }
@@ -53,5 +59,6 @@ class BackgroundWorker: NSObject {
     
     @objc private func runBlock() {
         block?()
+	block = nil
     }
 }
